core/core_c.h

```diff
@@ -39,12 +39,29 @@ typedef enum kernel_type_t {
   BUSY_WAIT,
   MEMORY_BOUND,
   COMPUTE_BOUND,
+  IO_BOUND,
+  LOAD_IMBALANCE
 } kernel_type_t;
 
+//-- add by Yuankun
+typedef struct kernel_arg_t{
+  char *scratch_ptr;
+  long num_src_input;
+  size_t scratch_bytes_per_task;
+
+  long max_power; // compute kernel parameter
+  long jump;  // memeory kernel parameter
```

Comment:
Let's move jump and max_power into kernel_t itself. For scratch_ptr and bytes, since those aren't available until later, let's make those arguments to Kernel::execute like I describe above.